wdl/transforms/draft3/src/main/scala/wdl/draft3/transforms/wdlom2wdl/WdlWriter.scala

```diff
@@ -0,0 +1,450 @@
+    override def toWdl(a: ExpressionElement) = a match {
+      case a: PrimitiveLiteralExpressionElement => a.toWdl
+      case a: StringExpression => "\"" + a.pieces.map(_.toWdl).mkString + "\""
+      case a: StringLiteral => "\"" + a.value + "\""
+      case a: ObjectLiteral =>
+        "object { " + a.elements.map { pair =>
+          pair._1 + ": " + expressionElementWriter.toWdl(pair._2)
+        }.mkString(", ") + " }"
+      case a: ArrayLiteral =>
+        "[" + a.elements.map(expressionElementWriter.toWdl).mkString(", ") + "]"
+      case a: MapLiteral =>
+        "{ " + a.elements.map { pair =>
+          expressionElementWriter.toWdl(pair._1) + ": " + expressionElementWriter.toWdl(pair._2)
+        }.mkString(", ") + " }"
+      case a: PairLiteral =>
+        s"(${expressionElementWriter.toWdl(a.left)}, ${expressionElementWriter.toWdl(a.right)})"
+      case a: UnaryOperation => a.toWdl
+      case a: BinaryOperation => a.toWdl
+      case a: TernaryIf =>
+        s"if ${expressionElementWriter.toWdl(a.condition)} then ${expressionElementWriter.toWdl(a.ifTrue)} else ${expressionElementWriter.toWdl(a.ifFalse)}"
+      case a: FunctionCallElement => a.toWdl
+      case a: IdentifierLookup => a.identifier
+      case a: IdentifierMemberAccess => a.toWdl
+      case a: ExpressionMemberAccess => s"${expressionElementWriter.toWdl(a.expression)}.${a.memberAccessTail.toList.mkString(".")}"
+      case a: IndexAccess => s"${expressionElementWriter.toWdl(a.expressionElement)}[${expressionElementWriter.toWdl(a.index)}]"
+    }
+  }
+
+  implicit val unaryOperationWriter: WdlWriter[UnaryOperation] = new WdlWriter[UnaryOperation] {
+    override def toWdl(a: UnaryOperation): String = a match {
+      case a: LogicalNot    => s"!(${a.argument.toWdl})"
+      case a: UnaryNegation => "-" + a.argument.toWdl
+      case a: UnaryPlus     => "+" + a.argument.toWdl
+    }
+  }
+
+  implicit val identifierMemberAccessWriter: WdlWriter[IdentifierMemberAccess] = new WdlWriter[IdentifierMemberAccess] {
+    override def toWdl(a: IdentifierMemberAccess): String = {
+      s"${a.first}.${a.second}" + (if (a.memberAccessTail.nonEmpty) {
+        "." + a.memberAccessTail.mkString(".")
+      } else {
+        ""
+      })
+    }
+  }
+
+  implicit val binaryOperationWriter: WdlWriter[BinaryOperation] = new WdlWriter[BinaryOperation] {
+    override def toWdl(a: BinaryOperation) = {
+      def sandwich(opString: String) = s"${a.left.toWdl} $opString ${a.right.toWdl}"
+
+      a match {
+        case _: LogicalOr           => sandwich("||")
+        case _: LogicalAnd          => sandwich("&&")
+        case _: Equals              => sandwich("==")
+        case _: NotEquals           => sandwich("!=")
+        case _: LessThan            => sandwich("< ")
+        case _: LessThanOrEquals    => sandwich("<=")
+        case _: GreaterThan         => sandwich("> ")
```

Comment:
Just OOI, is the extra space here deliberate?

Reply:
Nope, removed